### DISCUSSION TITLE
BIOS files handling for emulators

### DISCUSSION THREAD

**Discussion by mintymoons** [lutris/lutris#5504] [General]
Quoting @strycore's opinion about handling BIOS files in Lutris:

>I think we should come up with a more generic solution for dealing with BIOS files. One that's not specific for Duckstation. The same BIOS files would be needed for the PSX Retroarch cores. And we need a way to handle BIOS files on a more generic level for other types of platforms (Amiga, Dreamcast, PS2, ....)

And agreeing with him.

What are your ideas o proposals do you have to handling BIOS files in the same directory to most of emulators (preferably all of them)?

**Comment by strycore** [comment]
What we know about BIOS files: 

* They're specific to a hardware platform rather than a runner
* The checksums for the BIOS files are known, valid BIOS files are required to match against those checksums
* Distributing direct links  to BIOS files can be frowned upon / legally questionable, even though we are talking about systems that are over a quarter of a century old for the most part.
* Finding packs of BIOS files is very easy, lutris just has to know how to interpret them and where to store them according to each runner

The FS-UAE runner has a start of an implementation of BIOS file detection. It's not finished enough to be useful but it shows an example of how to keep track of BIOS files.
Ideally, a user would only need to provide a folder with all bios files with no particular structure of naming scheme and lutris should be able to scan the contents of it and detect known BIOS files. 
When we scan the ROM folder, we can build a mapping of platform => bios files, which we can use to copy BIOS files to the appropriate folder for a runner.  

We should take the region into account and add an option to set it in the runner if that's not already done properly by the runner (I think PCSX2 will just pick the correct region based on the game played)

MAME may not use the same ROM files as other runners

We need to have a way to map retroarch cores to platforms.

We can get started on a few platforms like Amiga, Playstation 1, Dreamcast and work from there.

**Reply by danieljohnson2** [reply]
Let's be clear though. It sounds like Lutris policy will be to *not* provide any BIOS files, nor links to them. Right?

**Reply by strycore** [reply]
That's right, but I don't see an issue distributing hashes to BIOS files. Which means it doesn't matter where they come from, Lutris will be able to handle them
